### PR TITLE
update code and add the possibility to run the jobs with multible cores

### PR DESCRIPTION
**Pull request by winterchristian**

### CODE REVIEW

File: gc_configs/freiburg.conf [hunk 1]
```diff
@@ -4,23 +4,24 @@ backend = condor
...
 workdir create = True
-workdir = __set__
+workdir = {workdir}
 
 [jobs]
 in flight = 1000
 wall time = 05:00:00
 memory = 3500
-jobs = __set__
+{jobs}
```
Comment (harrypuuter) [inline gc_configs/freiburg.conf:13]: why is this different than the other arguments ?

Comment (harrypuuter) [reply gc_configs/freiburg.conf:13]: not `jobs = {jobs}`

File: helpers.py [hunk 2]
```diff
@@ -4,10 +4,26 @@
...
+        "total_events",
+        "cpus_per_job",
+        "events_per_job",
+        "accounting_group",
+    ]
+    for key in dict.keys():
+        if key not in needed_keys:
+            return False
+
+    if len(dict.keys()) < 8:
```
Comment (harrypuuter) [inline helpers.py]: `if len(dict.keys()) < len(needed_keys):`

File: helpers.py [hunk 3]
```diff
@@ -28,109 +44,142 @@ def createFolder(path):
...
-    setupfile.write("seed=$(($(date +%s) % 100 + 1)) \n")
-    setupfile.write("cd {workdir} \n".format(workdir=os.path.join(workdir)))
-    setupfile.write("scram p {cmssw} \n".format(cmssw=version))
-    setupfile.write("cd {cmssw}/src \n".format(cmssw=os.path.join(version)))
-    setupfile.write("eval `scram runtime -sh` \n \n")
-
-    initfile.write("source /cvmfs/cms.cern.ch/cmsset_default.sh \n")
+    setupfile.write(
+        f"""
+export SCRAM_ARCH={arch}
```
Comment (harrypuuter) [inline helpers.py]: indent for the looks

File: request_config.yaml [hunk 4]
```diff
@@ -1,273 +1,417 @@
...
+  label: "ttbar_sim_ul18_summer20"
+  input_files: "/work/cwinter/embedding/MC_UL_production/MC_prod/LocalMonteCarloProduction/ttbar_lhegen/ttbar_ul18_summer20_lhegen.dbs"
+  workdir: "ttbar_sim"
+  output_path: "root://cmsxrootd-kit.gridka.de:1094//store/user/cwinter/ul_embedding/MC_UL_production/ttbar_sim"
+  batch_system: "freiburg"
+  total_events: 2000000
+  events_per_job: 1000
+  cpus_per_job: 4
+  accounting_group: "cms.production"
+# ttbar_premix_cwinter:
```
Comment (harrypuuter) [inline request_config.yaml]: why commented out ?

**Review by harrypuuter — CHANGES_REQUESTED** [review]
consider to split your production in a separate config

**Comment by harrypuuter** [comment]
is this ready for merging ? :)

**Comment by harrypuuter** [comment]
@winterchristian Is this ready for merging?

**Comment by harrypuuter** [comment]
ping

File: gc_configs/freiburg.conf (hunk 1 above)
Comment (winterchristian) [reply gc_configs/freiburg.conf:13]: In the first step of the simulation chain we set this option `jobs` to limit the number of events per job. Afterward we don't need this setting anymore, as one file per job is used for the rest of the simulation processing chain.

**Pull request merged**